ambry-commons/src/main/java/com/github/ambry/mysql/MySqlDataAccessor.java

```diff
@@ -123,6 +124,43 @@ private void initializeDriver(String url) throws SQLException {
     }
   }
 
+  /**
+   * Enables or disables auto commit on current active {@link Connection}.
+   * @param enable {@code true} to enable auto commit and {@code false} to disable auto commit.
+   * @throws SQLException
+   */
+  public synchronized void setAutoCommit(boolean enable) throws SQLException {
+    if (activeConnection != null && activeConnection.isValid(5)) {
+      activeConnection.setAutoCommit(enable);
+    } else {
+      logger.debug("Connection to MySql DB is not available");
```

Comment:
This method is called as the first step in a transaction.  I would expect it to establish a connection if one does not exist.